src/SimpleDetailsList.tsx

```diff
@@ -87,16 +89,16 @@ export class SimpleDetailsList extends React.Component<any, IDetailsListBasicExa
     return (
       <div className='ms-DetailsListBasicExample'>
         <Toggle
-          isToggled={ isGridVisible }
+          checked={ isGridVisible }
           onChanged={ isToggled => this.setState({ isGridVisible: isToggled }) }
           label='Grid visible'
           onText='On'
-          offText='Off' 
+          offText='Off'
           />
 
         <CommandBar items={ this._getCommandItems() } />
 
-        <div style={ isGridVisible ? null : { display: "none" } }>
+        { isGridVisible && (

```

Comment:
The solution is easy because the problem was easy on this repro. Actually visibility might not be available in the component, it's applied in a parent component and propagating it all the way down sounds like pollution.